### PR TITLE
[wasm] fix prefetch in samples

### PR DESCRIPTION
**Pull request by pavelsavara**
- only browser-advanced and browser-bench have preload optimizations now
- modulepreload is not yet supported by FireFox, script tag type='module' works better for dependencies
- preload and prefetch works well enough

Setting [Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) header on the server responses may further improve perf.

### CODE REVIEW

File: src/mono/sample/wasm/browser-advanced/index.html [hunk 1]
```diff
@@ -7,18 +7,17 @@
...
-  <link rel="modulepreload" href="./dotnet.js" />
+  <script type='module' src="./main.js"></script>
+  <script type='module' src="./dotnet.js"></script>
   <link rel="preload" href="./mono-config.json" as="fetch" crossorigin="anonymous">
-  <link rel="preload" href="./dotnet.wasm" as="fetch" crossorigin="anonymous">
-  <link rel="preload" href="./icudt.dat" as="fetch" crossorigin="anonymous">
-  <link rel="preload" href="./managed/System.Private.CoreLib.dll" as="fetch" crossorigin="anonymous">
+  <link rel="prefetch" href="./dotnet.wasm" as="fetch" crossorigin="anonymous">
+  <link rel="prefetch" href="./icudt.dat" as="fetch" crossorigin="anonymous">
+  <link rel="prefetch" href="./managed/System.Private.CoreLib.dll" as="fetch" crossorigin="anonymous">
```
Comment (radekdoulik) [inline src/mono/sample/wasm/browser-advanced/index.html:15]: Shouldn't the link element end with `/>`?

**Review by radekdoulik — APPROVED** [review]
LGTM. Let see how it will look in graphs.

I was playing with it locally too and for me this sequence worked  best in firefox:
```
@@ -5,14 +5,14 @@

 <head>
   <title>App task</title>
-  <meta charset="UTF-8">
-  <meta name="viewport" content="width=device-width, initial-scale=1.0">
-  <link rel="modulepreload" href="./frame-main.js" />
-  <link rel="modulepreload" href="./dotnet.js" />
-  <link rel="preload" href="./mono-config.json" as="fetch" crossorigin="anonymous">
-  <link rel="preload" href="./dotnet.wasm" as="fetch" crossorigin="anonymous">
-  <link rel="preload" href="./icudt.dat" as="fetch" crossorigin="anonymous">
-  <link rel="preload" href="./managed/System.Private.CoreLib.dll" as="fetch" crossorigin="anonymous">
+  <meta charset="UTF-8"/>
+  <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
+  <link rel="preload" href="./frame-main.js" as="fetch" crossorigin="anonymous"/>
+  <link rel="preload" href="./dotnet.js" as="fetch" crossorigin="anonymous"/>
+  <link rel="preload" href="./mono-config.json" as="fetch" crossorigin="anonymous"/>
+  <link rel="preload" href="./dotnet.wasm" as="fetch" crossorigin="anonymous"/>
+  <link rel="preload" href="./icudt.dat" as="fetch" crossorigin="anonymous"/>
+  <link rel="preload" href="./managed/System.Private.CoreLib.dll" as="fetch" crossorigin="anonymous"/>
 </head>

 <body>
```
We can try to change it after few days and see how it will compare.

It measured 503.2500ms (after) vs 571.2727ms(before)

File: src/mono/sample/wasm/browser-advanced/index.html (hunk 1 above)
Comment (maraf) [reply src/mono/sample/wasm/browser-advanced/index.html:15]: It's optional in html5

**Comment by pavelsavara** [comment]
>  `<link rel="preload" href="./dotnet.js" as="fetch" crossorigin="anonymous"/>`

That doesn't work for me on FF, the file is downloaded twice. Even if I fix the `as="script" `. I guess script is not a module.
image



I will move scripts into header for all other samples too, based on @maraf's feedback

**Review by maraf — APPROVED** [review]

**Pull request merged**